### ISSUE TITLE
Support of MySQL database selection

### ISSUE THREAD

**Issue by Tiross** [sqlfluff/sqlfluff#902]
Hello, 

As I use many MySQL procedures, I used to stored them in SQL file in order to import them into my server when needed. 

Every files start with the 2 same lines 

```sql 
CREATE DATABASE IF NOT EXISTS my_db;
USE my_db;
```

These 2 lines cause a `PSR` error, SQLFluff does not parse `USE my_db`.

```
echo 'CREATE DATABASE IF NOT EXISTS my_db; USE my_db;' | sqlfluff lint --dialect mysql -
== [stdin] FAIL
L:   1 | P:  38 |  PRS | Found unparsable section: 'USE my_db'
```

[Online fluff](http://sqlfluff-online.herokuapp.com/fluffed?sql=H4sIAFKVX2AC_wsNdlXIrYxPSbLm5QIAhvctiwwAAAA%3D&dialect=mysql)

Is it possible to support `USE` ? 

Thank you

**Comment by dmateusp** [comment]
hi @Tiross, thanks for raising the issue!

I'm going to take that on, I'd like to also support the variations from Snowflake:

https://docs.snowflake.com/en/sql-reference/sql/use.html

```sql
USE ROLE <name>

USE WAREHOUSE <name>

USE [ DATABASE ] <name>

USE [ SCHEMA ] [<db_name>.]<name>
```

**Comment by dmateusp** [comment]
Actually we already have logic to support the Snowflake USE statements, so I'm going to extract it into our base (ANSI) dialect
